dace/codegen/compiled_sdfg.py

```diff
@@ -366,21 +366,56 @@ def _get_error_text(self, result: Union[str, int]) -> str:
             return result
 
     def __call__(self, *args, **kwargs):
-        # Update arguments from ordered list
-        if len(args) > 0 and self.argnames is not None:
-            kwargs.update({aname: arg for aname, arg in zip(self.argnames, args)})
+        """This function forwards the Python call to the complied `C` code.
 
-        try:
-            argtuple, initargtuple = self._construct_args(kwargs)
+        The positional arguments (`args`) are expected to be in the same order as
+        `argnames` (which comes from `arg_names` in the source `SDFG`).
+        This function will roughly do the following steps:
+        - bringing the arguments in the order dictated by the C callable.
+        - perfrom some basic checks on the arguments.
+        - transfroms `ndarray`s to their `C` equivalent.
 
+        If you are know what you are doing you can also use `_fast_call()` which
+        allows to bypass these operations and call the extension directly.
+        """
+        if self.argnames is None and len(args) != 0:
+            raise KeyError(f"Passed positional arguments to an SDFG that does not accept them.")
+        elif len(args) > 0 and self.argnames is not None:
+            kwargs.update(
+                # `_construct_args` will handle all of its argument as kwargs.
+                {aname: arg for aname, arg in zip(self.argnames, args)}
+            )
+        argtuple, initargtuple = self._construct_args(kwargs)   # Missing arguments will be detected here.
+                                                                #  Return values cached in `self._lastargs`.
+        return self._fast_call(argtuple, initargtuple)
```

Comment:
I am not sure what you mean by that. since `_fast_call()` returns the result of `_convert_return_values()` `__call__()` will return the same. So I do not see a need to call it again.